### PR TITLE
Organizing the index

### PR DESCRIPTION
**Pull request by jakiestfu**
- Trying to reduce the use of process.env in order to boot the game
- Simplified API for game state interaction/potential persistence

### CODE REVIEW

File: src/game.ts [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+      store.dispatch(winnerDeclared({ winner: sovereign }))
+    }
+  })
+}
+
+export const gameClock: () => void = () => {
+  const serverTick = () => store.dispatch(tick())
+  serverTick()
+  setInterval(serverTick, 6000)
+}
```
Comment (brian-gates) [inline src/game.ts:171]: 😍

File: src/index.ts [hunk 2]
```diff
@@ -1,155 +1,61 @@
...
       )
-    }
-    console.timeEnd(interaction.commandName)
-  })
-  client.on('error', (e) => {
-    console.error('Discord client error!', e)
-  })
+      if (didStateChange) {
+        // Persist to DB
+      }
```
Comment (brian-gates) [inline src/index.ts:40]: Isn't this essentially what redux-persist does?

File: src/index.ts [hunk 3]
```diff
@@ -1,155 +1,61 @@
...
+    token,
+    clientId,
+    channelId,
+    onBeforeInteraction: async (gameState) => {
+      // Keep track of what the game state is prior to the users interaction
+      lastState = gameState
+    },
+    onAfterInteraction: async (gameState) => {
+      const didStateChange = Boolean(
+        Object.keys(diff(lastState, gameState)).length
```
Comment (brian-gates) [inline src/index.ts:36]: How expensive is deep diffing the whole state? Given that this happens every interaction, we'll need to be cognizant of the load here. 

I like the PoC but I think we should avoid merging this to main until we're ready to full take it through to persistence.

File: src/index.ts (hunk 2 above)
Comment (jakiestfu) [reply src/index.ts:40]: Yes, I'm going to change this after we'd spoken about the site design/use

**Review by brian-gates — APPROVED** [review]

**Pull request merged**